Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+
+        self.setPropertySettings('InstrumentParameterFile', scan)
+
+    def _figure_measurement_method(self, ws):
+        """Figures out the measurement method based on the structure of the input files."""
+        entries_per_numor = mtd[ws].getNumberOfEntries() / len(self.getPropertyValue('Run').split(','))
+        if entries_per_numor == 10:
+            self._method_data_structure = '10p'
+        elif entries_per_numor == 6:
+            self._method_data_structure = 'XYZ'
+            if self._user_method == '10p':
+                raise RunTimeError("The provided data cannot support 10-point measurement component separation.")
+        elif entries_per_numor == 2:
+            self._method_data_structure = 'Uniaxial'
+            if self._user_method == '10p':
+                raise RunTimeError("The provided data cannot support 10-point measurement component separation.")
+            if self._user_method == 'XYZ':
+                raise RunTimeError("The provided data cannot support XYZ measurement component separation.")
+        else:
+            if self.getPropertyValue("ProcessAs") not in ['Beam', 'Transmission']:
+                raise RuntimeError("The analysis options are: Uniaxial, XYZ, and 10p. "
+                                   + "The provided input does not fit in any of these measurement types.")
+
+    def _merge_polarisations(self, ws, average_detectors=False):
+        """Merges workspaces with the same polarisation inside the provided WorkspaceGroup either by using SumOverlappingTubes
+        or averaging entries for each detector depending on the status of the sumOverDetectors flag."""
+        pol_directions = set()
+        numors = set()
+        for name in mtd[ws].getNames():
+            last_underscore = name.rfind("_")
+            numors.add(name[:last_underscore])
+            pol_directions.add(name[last_underscore+1:])
+        if len(numors) > 1:
+            names_list = []
+            for direction in sorted(list(pol_directions)):
+                name = '{0}_{1}'.format(ws, direction)
+                list_pol = []
+                for numor in numors:
+                    if average_detectors:
+                        try:
+                            Plus(LHSWorkspace=name, RHSWorkspace=mtd[numor + '_' + direction], OutputWorkspace=name)
+                        except ValueError:
+                            CloneWorkspace(InputWorkspace=mtd[numor + '_' + direction], OutputWorkspace=name)
+                    else:
+                        list_pol.append('{0}_{1}'.format(numor, direction))
+                if average_detectors:
+                    Divide(LHSWorkspace=name, RHSWorkspace=CreateSingleValuedWorkspace(len(numors)), OutputWorkspace=name)
+                else:
+                    SumOverlappingTubes(','.join(list_pol), OutputWorkspace=name,
+                                        OutputType='1D', ScatteringAngleBinning=0.5, Normalise=True, HeightAxis='-0.1,0.1')
```

Comment:
Should the binning be a user input? Do you need the height axis for 1D?

Reply:
Scattering angle binning may be exposed to the user, yes. The height axis needs to be defined, or else we get the following error: `No detector_for_height_axis parameter for this instrument. Please enter a value for the HeightAxis parameter`